test/suite/wttest.py

```diff
@@ -923,6 +923,14 @@ def runit_decorator(func):
     else:
         return runit_decorator
 
+# Override a test's setUp function to instead skip and report the reason for skipping
+def register_skipped_test(test, hook, skip_reason):
+
+    def _skip_test(self):
+        raise unittest.SkipTest(f"{test} for hook {hook}: {skip_reason}")
+
+    setattr(test, "setUp", lambda: _skip_test(test))
```

Comment:
This warrants extra scrutiny since we're overriding the function to be executed at runtime. Throwing `unittest.SkipTest` is [the correct way to do this](https://docs.python.org/3/library/unittest.html#unittest-skipping), but this feels a bit messy to me.